src/gt4py/cartesian/gtscript.py

```diff
@@ -159,6 +160,26 @@ def _parse_annotation(arg, annotation):
     return original_annotations
 
 
+_ENUM_REGISTER: dict[str, object] = {}
+"""Register of IntEnum that will be available to parsing in stencils. Register
+with @gtscript.enum()"""
+
+
+def enum(class_: type[IntEnum]):
+    class_name = class_.__name__
+    if class_name in _ENUM_REGISTER:
+        raise ValueError(
+            f"Cannot register @gtscript.enum {class_name} as a class"
+            "with the same name is already registered."
```

Comment:
I'd suggest a slightly less technical error message since this is user facing. The fact that we keep enums in a registry is an implementation detail and shouldn't need to bother users.

```suggestion
            f"Enum names must be unique. @gtscript.enum {class_name} is already taken."
```